lib/puppet/provider/package/te_agent_bin.rb

```diff
@@ -14,16 +14,41 @@
   has_feature :versionable
   has_feature :install_options
 
+  confine :osfamily => [ :RedHat ]
+  confine :exists   => "/etc/init.d/twdaemon"
+
   def self.prefetch(packages)
     packages.each do |name, pkg|
       version = get_version(pkg)
-      pkg.provider = new({:ensure => version, :name => name, :provider => :te_agent_bin}) if version
+      pkg.provider = new({:name => name, :ensure => version, :provider => :te_agent_bin}) if version
     end
   end
 
+  def self.instances
+    paths = []
+    versions = []
+    File.open('/etc/init.d/twdaemon').each_line do |r|
```

Comment:
This will error out if the agent isn't installed.

Reply:
fixed with latest commit : https://github.com/sudodevnull/puppet-te_agent/commit/355c65b5c213ba2105f63c6525a271dd3319cdd5